### PR TITLE
fix too restrictive regex on pathpart

### PR DESCRIPTION
**Pull request by helloworlddan**
The RegEx was too restrictive when being used with proxy variables. The
\+ in {proxy+} is valid and should not throw an exception.

Signed-off-by: Daniel Stamer <daniel.stamer@cloudreach.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license.

### CODE REVIEW

File: packages/@aws-cdk/aws-apigateway/lib/resource.ts [hunk 1]
```diff
@@ -140,7 +140,7 @@ function validateResourcePathPart(part: string) {
         part = part.substr(1, part.length - 2);
     }
 
-    if (!/^[a-zA-Z0-9\.\_\-]+$/.test(part)) {
+    if (!/^[a-zA-Z0-9\.\_\-]+[\+]?$/.test(part)) {
         throw new Error(`Resource's path part only allow a-zA-Z0-9._- and curly braces at the beginning and the end: ${part}`);
```
Comment (eladb) [inline packages/@aws-cdk/aws-apigateway/lib/resource.ts]: Can you please update the message?

**Comment by RomainMuller** [comment]
Thanks for the contribution!

I feel that it would be best if we can make sure the `+` is only allowed within a path parameter fragment, before the closing `}`. It otherwise makes the validation permit many invalid uses... Would you be able to change your code to this effect?

**Comment by eladb** [comment]
@RomainMuller just to make sure I understand, so we just want to allow either `}` or `+}` as suffixes, correct instead of generally allowing `+` as a character within the path part. Correct?

**Comment by RomainMuller** [comment]
@eladb Correct.

**Comment by eladb** [comment]
Hi @daniceman, when you are ready to merge this, let me know. I will take care of getting it aligned with "master". We've converted our codebase from 4 to 2 spaces, so there's a bit of juggling to do to help it land

File: packages/@aws-cdk/aws-apigateway/lib/resource.ts (hunk 1 above)
Comment (daniceman) [reply packages/@aws-cdk/aws-apigateway/lib/resource.ts]: @eladb I updated the message to account for the optional trailing '+'.

**Comment by daniceman** [comment]
> Thanks for the contribution!
> 
> I feel that it would be best if we can make sure the `+` is only allowed within a path parameter fragment, before the closing `}`. It otherwise makes the validation permit many invalid uses... Would you be able to change your code to this effect?

The regex checks for an optional _trailing_ '+', so I think all matches should be valid. Am I understanding this incorrectly?

**Comment by daniceman** [comment]
@eladb 
> Hi @daniceman, when you are ready to merge this, let me know. I will take care of getting it aligned with "master". We've converted our codebase from 4 to 2 spaces, so there's a bit of juggling to do to help it land

I re-tabed the source to 2 spaces.

**Comment by Rukenshia** [comment]
@daniceman the trailing + is only allowed inside of parameters as far as I know, not in normal paths

**Comment by daniceman** [comment]
@Rukenshia 
> @daniceman the trailing + is only allowed inside of parameters as far as I know, not in normal paths

I merged in your code & tests. Now the optional '+' is explicitly stripped before the path is RegEx matched.

@RomainMuller Could you quickly check if @Rukenshia fixed your concerns?

**Comment by daniceman** [comment]
@eladb I think this PR is ready to merge now.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,9 @@
+## [UNRELEASED]
+
+### Bug Fixes
+
+* **aws-apigateway:** fix validation of resources failing for proxy resource paths (`{proxy+}`)
```
Comment (eladb) [inline CHANGELOG.md]: We haven’t updated our CONTIBUTING guide yet, but this is not needed anymore. Instead just make sure your squashed commit to master adheres to conventionalcommits. Sorry for the trouble.

**Pull request merged**